lib/runners/harness.rb

```diff
@@ -91,5 +94,33 @@ def handle_error(exn)
         trace_writer.error "#{exn.message} (#{exn.class})"
       end
     end
+
+    def remove_ignored_files(config)
+      ignores = config.ignore
+      return if ignores.empty?
+
+      trace_writer.message("Deleting specified files via the `ignore` option...") do
+        with_gitignore(ignores) do |list|
+          FileUtils.rm_rf(list)
+          trace_writer.message list.map { |f| f.relative_path_from(working_dir) }.join("\n")
+        end
+      end
+    end
+
+    def with_gitignore(ignores)
+      gitignore = (working_dir / ".gitignore")
+
+      backup = gitignore.file? ? gitignore.read : nil
+      gitignore.write(ignores.join("\n"))
+
+      shell = Shell.new(current_dir: working_dir, trace_writer: trace_writer, env_hash: {})
+      shell.capture3!("git", "init")
+      stdout, = shell.capture3("git", "check-ignore", *working_dir.glob("**/*", File::FNM_DOTMATCH).map(&:to_s))
```

Comment:
[suggestion] What about checking the exit status for safety like this?

```suggestion
      # @see https://git-scm.com/docs/git-check-ignore#_exit_status
      stdout, = shell.capture3!("git", "check-ignore", *working_dir.glob("**/*", File::FNM_DOTMATCH).map(&:to_s), is_success: ->(s) { [0, 1].include?(s.exitstatus) })
```

See https://git-scm.com/docs/git-check-ignore#_exit_status

Reply:
[suggestion] What about adding `trace_command_line: false, trace_stdout: false` also here?

Reply:
OK, I do it. I think to check whether the exit status is `128` or not seems better.

Reply:
3ea37cd

Reply:
What do you think about https://github.com/sider/runners/pull/798#discussion_r386255007?

Reply:
66accd5